From 684c69bbdf05021007d18f969be9f4609fe5e72b Mon Sep 17 00:00:00 2001
From: KayYCHuang-wiwynn <kay.yc.huang.wiwynn@gmail.com>
Date: Thu, 24 Jul 2025 15:54:40 +0000
Subject: [PATCH] Add event log type for post timeout

[Task Description]
- Related to YV4T1M-1993
- Implement SEL to record events when system fails to complete POST within 20 minutes after DC power on or platform reset

[Motivation]
- To notify users when POST does not complete in time

[Design]
- Add new event type "Post-Timeout"

Signed-off-by: KayYCHuang-wiwynn <kay.yc.huang.wiwynn@gmail.com>
---
 oem/meta/libpldmresponder/oem_meta_file_io_type_event_log.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/oem/meta/libpldmresponder/oem_meta_file_io_type_event_log.cpp b/oem/meta/libpldmresponder/oem_meta_file_io_type_event_log.cpp
index 3af8216..99c09c9 100644
--- a/oem/meta/libpldmresponder/oem_meta_file_io_type_event_log.cpp
+++ b/oem/meta/libpldmresponder/oem_meta_file_io_type_event_log.cpp
@@ -114,6 +114,7 @@ int EventLogHandler::write(const message& eventData)
         " OS LOAD WDT PWR_DOWN",
         " OS LOAD WDT PWR_CYCLE",
         " MTIA VR FAULT",
+        " Post-Timeouted",
     });
 
     static constexpr auto pmic_event_type = std::to_array({
-- 
2.25.1

